### PR TITLE
Make users know about telemetry via `startupWarnings`

### LINKED ISSUE #1328: Make users know about telemetry via `startupWarnings`
### What should be done?

When telemetry is in an undecided state, we currently log about using normal process log. We should also log about it in `getLog`s `startupWarnings`. There, information about telemetry will be visible to `mongosh` users:

```
-> task mongosh
task: [mongosh] docker-compose exec mongodb mongosh mongodb://host.docker.internal:27017/test?heartbeatFrequencyMS=300000 --verbose --eval 'disableTelemetry()' --shell

Current Mongosh Log ID:	63594889f1f8632e18dbba0c
Connecting to:		mongodb://host.docker.internal:27017/test?heartbeatFrequencyMS=300000&directConnection=true&appName=mongosh+1.6.0
Using MongoDB:		6.0.42
Using Mongosh:		1.6.0

For mongosh info see: https://docs.mongodb.com/mongodb-shell/


To help improve our products, anonymous usage data is collected and sent to MongoDB periodically (https://www.mongodb.com/legal/privacy-policy).
You can opt-out by running the disableTelemetry() command.

------
   The server generated these startup warnings when booting
   2022-10-26T14:47:37.67Z: Powered by 🥭 FerretDB v0.5.4-129-ge3c228b7 and PostgreSQL 15.0.
   2022-10-26T14:47:37.67Z: Please star us on GitHub: https://github.com/FerretDB/FerretDB
   2022-10-26T14:47:37.67Z: Telemetry state undecided, waiting 1h0m0s before the first report. Read more about FerretDB telemetry at https://beacon.ferretdb.io
------
```

We should implement that last line: "Telemetry state undecided..."

Test should be added.

Extracting common part of `pg.Handler.MsgGetLog` and `tigris.Handler.MsgGetLog` into some common function or similar refactoring is not in the scope.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Closes #1328.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/util/state/state.go [hunk 1]
```diff
@@ -26,8 +26,9 @@ import (
 
 // State represents FerretDB process state.
 type State struct {
-	UUID      string `json:"uuid"`
-	Telemetry *bool  `json:"telemetry,omitempty"` // nil for undecided
+	UUID           string        `json:"uuid"`
+	Telemetry      *bool         `json:"telemetry,omitempty"` // nil for undecided
+	UndecidedDelay time.Duration `json:"undecided_delay,omitempty"`
```
Comment (rumyantseva) [inline internal/util/state/state.go:31]: I added it to the state as this info is needed in the warning text. Let me know if you have a better idea.

Comment (AlekSi) [reply internal/util/state/state.go:31]: Let’s remove it from the text instead

Comment (rumyantseva) [reply internal/util/state/state.go]: fixed

File: integration/commands_diagnostic_test.go [hunk 2]
```diff
@@ -125,6 +125,28 @@ func TestCommandsDiagnosticGetLog(t *testing.T) {
...
+	var actual bson.D
+	err := collection.Database().RunCommand(ctx, command).Decode(&actual)
+	require.NoError(t, err)
+
+	m := actual.Map()
+	log := m["log"].(bson.A)
+	assert.Equal(t, 3, len(log))
+
+	// check that the last line contains info about telemetry being undecided
+	assert.Contains(t, log[2].(string), "Telemetry state undecided")
```
Comment (AlekSi) [inline integration/commands_diagnostic_test.go:147]: That test assumes that telemetry state is undecided, but nothing actually ensures that. I think we should check it in common setup

File: internal/handlers/pg/msg_getlog.go [hunk 3]
```diff
@@ -83,14 +83,27 @@ func (h *Handler) MsgGetLog(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
+		startupWarnings := []string{
+			"Powered by FerretDB " + mv.Version + " and PostgreSQL " + pv + ".",
+			"Please star us on GitHub: https://github.com/FerretDB/FerretDB.",
+		}
+
+		state := h.StateProvider.Get()
+		if state.Telemetry == nil {
+			startupWarnings = append(
+				startupWarnings,
+				"The telemetry state is undecided; the first report will be sent soon. "+
```
Comment (chilagrow) [inline internal/handlers/pg/msg_getlog.go:99]: Just want to double check, not showing how long we wait until the first report is sent isn't an issue here and tigris.

Comment (rumyantseva) [reply internal/handlers/pg/msg_getlog.go:99]: To implement this, we need to know the delay. My first implementation contained the delay being present in the `State`, but @AlekSi said it's better not to have delay time in the warning than to have it stored in the `State`. So, this is why :)

File: internal/handlers/pg/msg_getlog.go [hunk 4]
```diff
@@ -83,14 +83,27 @@ func (h *Handler) MsgGetLog(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		))
 
 	case "startupWarnings":
+		// TODO telementry
```
Comment (w84thesun) [inline internal/handlers/pg/msg_getlog.go:86]:
```suggestion
		// TODO telemetry
```

**Review by w84thesun — CHANGES_REQUESTED** [review]
Overall looks good, small fix added.

**Review by noisersup — APPROVED** [review]
I understand the logic and it looks good to me, I didn't find any other suggestions so I leave approve

File: integration/commands_diagnostic_test.go (hunk 2 above)
Comment (rumyantseva) [reply integration/commands_diagnostic_test.go]: Fixed

File: integration/setup/common.go [hunk 5]
```diff
@@ -93,10 +94,13 @@ func SkipForPostgresWithReason(tb testing.TB, reason string) {
...
 // and returns listening port number.
-func setupListener(tb testing.TB, ctx context.Context, logger *zap.Logger) int {
+func setupListener(tb testing.TB, ctx context.Context, logger *zap.Logger) (*state.Provider, int) {
 	tb.Helper()
 
-	p, err := state.NewProvider("")
+	stateFile, err := os.CreateTemp("", "test-*.state")
+	require.NoError(tb, err)
+
+	p, err := state.NewProvider(stateFile.Name())
```
Comment (AlekSi) [inline integration/setup/common.go:103]:
```suggestion
	p, err := state.NewProvider("")
```

We don't have to persist anything in those tests

File: internal/handlers/pg/msg_getlog.go (hunk 3 above)
Comment (AlekSi) [reply internal/handlers/pg/msg_getlog.go:98]: I'm not 100% sure what state _is_, but I'm sure we don't want to be a grab-bag of everything, including CLI flags.

File: integration/commands_diagnostic_test.go [hunk 6]
```diff
@@ -125,6 +127,64 @@ func TestCommandsDiagnosticGetLog(t *testing.T) {
...
+			res := setup.SetupWithOpts(t, &setup.SetupOpts{
+				DatabaseName: "admin" + name,
+			})
+
+			ctx, collection := res.Ctx, res.Collection
+
+			err := res.StateProvider.Update(func(s *state.State) { s.Telemetry = tc.telemetry })
+			require.NoError(t, err)
+
+			// Check that state was stored correctly.
```
Comment (AlekSi) [inline integration/commands_diagnostic_test.go:165]: We have separate tests for that

File: integration/setup/common.go (hunk 5 above)
Comment (rumyantseva) [reply integration/setup/common.go:103]: @AlekSi but will it work? I mean, will I be able to change the state through `Update` and expect it to be persistent if there is no state file?

Comment (AlekSi) [reply integration/setup/common.go:103]: Yes, there are tests for that

**Review by chilagrow — DISMISSED** [review]

File: integration/commands_diagnostic_test.go (hunk 6 above)
Comment (rumyantseva) [reply integration/commands_diagnostic_test.go]: fixed

**Review by noisersup — CHANGES_REQUESTED** [review]
Overall looks good to me, although mongodb tests are failing and they don't look flaky

**Comment by rumyantseva** [comment]
@noisersup nice catch!

File: integration/commands_diagnostic_test.go [hunk 7]
```diff
@@ -125,6 +127,60 @@ func TestCommandsDiagnosticGetLog(t *testing.T) {
 	}
 }
 
+func TestCommandsDiagnosticGetLogTelemetryReported(t *testing.T) {
```
Comment (rumyantseva) [inline integration/commands_diagnostic_test.go:130]: @AlekSi should this test actually be in `dance` as startup warnings for MongoDB look different? Or should we do smth like `skipForMongo` here?

**Review by w84thesun — DISMISSED** [review]

File: integration/commands_diagnostic_test.go (hunk 7 above)
Comment (AlekSi) [reply integration/commands_diagnostic_test.go:130]: Let's cut that test for now

**Review by AlekSi — COMMENTED** [review]
https://github.com/FerretDB/FerretDB/pull/1336#discussion_r1010115962

**Review by chilagrow — APPROVED** [review]

**Review by noisersup — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Pull request merged**